x-pack/plugin/runtime-fields/src/main/java/org/elasticsearch/xpack/runtimefields/mapper/ScriptBooleanMappedFieldType.java

```diff
@@ -0,0 +1,202 @@
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+package org.elasticsearch.xpack.runtimefields.mapper;
+
+import org.apache.lucene.search.MatchNoDocsQuery;
+import org.apache.lucene.search.Query;
+import org.apache.lucene.util.BytesRef;
+import org.elasticsearch.common.lucene.search.Queries;
+import org.elasticsearch.common.time.DateMathParser;
+import org.elasticsearch.index.mapper.BooleanFieldMapper;
+import org.elasticsearch.index.query.QueryShardContext;
+import org.elasticsearch.script.Script;
+import org.elasticsearch.search.DocValueFormat;
+import org.elasticsearch.search.lookup.SearchLookup;
+import org.elasticsearch.xpack.runtimefields.BooleanScriptFieldScript;
+import org.elasticsearch.xpack.runtimefields.fielddata.ScriptBooleanFieldData;
+import org.elasticsearch.xpack.runtimefields.query.BooleanScriptFieldExistsQuery;
+import org.elasticsearch.xpack.runtimefields.query.BooleanScriptFieldTermQuery;
+
+import java.time.ZoneId;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Supplier;
+
+public class ScriptBooleanMappedFieldType extends AbstractScriptMappedFieldType {
+    private final BooleanScriptFieldScript.Factory scriptFactory;
+
+    ScriptBooleanMappedFieldType(String name, Script script, BooleanScriptFieldScript.Factory scriptFactory, Map<String, String> meta) {
+        super(name, script, meta);
+        this.scriptFactory = scriptFactory;
+    }
+
+    @Override
+    protected String runtimeType() {
+        return BooleanFieldMapper.CONTENT_TYPE;
+    }
+
+    @Override
+    public Object valueForDisplay(Object value) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public DocValueFormat docValueFormat(String format, ZoneId timeZone) {
+        if (format != null) {
+            throw new IllegalArgumentException("Field [" + name() + "] of type [" + typeName() + "] does not support custom formats");
+        }
+        if (timeZone != null) {
+            throw new IllegalArgumentException("Field [" + name() + "] of type [" + typeName() + "] does not support custom time zones");
+        }
+        return DocValueFormat.BOOLEAN;
+    }
+
+    @Override
+    public ScriptBooleanFieldData.Builder fielddataBuilder(String fullyQualifiedIndexName, Supplier<SearchLookup> searchLookup) {
+        return new ScriptBooleanFieldData.Builder(name(), leafFactory(searchLookup.get()));
+    }
+
+    private BooleanScriptFieldScript.LeafFactory leafFactory(SearchLookup searchLookup) {
+        return scriptFactory.newFactory(script.getParams(), searchLookup);
+    }
+
+    @Override
+    public Query existsQuery(QueryShardContext context) {
+        checkAllowExpensiveQueries(context);
+        return new BooleanScriptFieldExistsQuery(script, leafFactory(context.lookup()), name());
+    }
+
+    @Override
+    public Query rangeQuery(
```

Comment:
I'm trying to mimic what we support in core. It works, so far as I can tell. It is kind of amazing we allow this at all, but we do!

Reply:
oh boy :)